lib/new_relic/agent/configuration/manager.rb

```diff
@@ -376,12 +376,13 @@ def new_cache
         end
 
         def log_config(direction, source)
-          # Just generating this log message (specifically calling
-          # flattened.inspect) is expensive enough that we don't want to do it
-          # unless we're actually going to be logging the message based on our
-          # current log level.
+          # Just generating this log message (specifically calling `flattened`)
+          # is expensive enough that we don't want to do it unless we're
+          # actually going to be logging the message based on our current log
+          # level, so use a `do` block.
           ::NewRelic::Agent.logger.debug do
-            "Updating config (#{direction}) from #{source.class}. Results: #{flattened.inspect}"
+            hash = flattened.delete_if { |k, _h| DEFAULTS.fetch(k, {}).fetch(:exclude_from_reported_settings, false) }
```

Comment:
There are 3 methods the manager offers that do similar things:

-`flattened`: This simply "flattens" the multidimensional `default_source.rb` `DEFAULTS` hash into a single dimensional hash. It is used internally by the agent for reading config values and so it should not have anything pruned from it.
- `to_collector_hash`: This method grabs the "flattened" hash, brings in the functionality offered by `DottedHash`, and then prunes away any config setting that has `:exclude_from_reported_settings` set to `true`. This data is used for submitting info to the collector, and so it can be pruned.
- `log_config`: This method grabs the "flattened" hash and prunes away any config setting that has `:exclude_from_reported_settings` set to `true`. This data is used for debug logging and so it can be pruned.

We can't prune anything in the `flattened` method because all of that data is needed by the agent.

We could possibly have the `log_config` method make use of the `to_collector_hash` method given that they both need to perform the same kind of pruning. I thought about this and gave it a try originally, but before pruning the contents of the hashes for these 2 methods are actually different.

I want this PR to prevent potentially sensitive info from being output but I stopped at making any change that would impact anything else. Given that `to_collector_hash` and `log_config` have slightly different hashes, I decided to leave them be with a bit of redundancy.

Reply:
Thank you for this breakdown! This is helpful context. Let's leave things as-is.